misc/tools.py

```diff
@@ -164,7 +162,7 @@ def checkmarks():
             pass
         continue
 
-    d = list(res)
+    d = res[:]
```

Comment:
Check that `res` is a list that needs to be copied and make sure it's not an iterator.